src/onegov/search/mixins.py

```diff
@@ -134,6 +136,139 @@ def es_tags(self):
         """ Returns a list of tags associated with this content. """
         return None
 
+    @property
+    def search_score(self):
+        """
+        the lower the score they higher the class type will be shown in search
+        results. Default is 10 (lowest)
+        """
+        return 10
+
+    def __lt__(self, other):
+        return self.search_score < other.search_score
+
+    @staticmethod
+    def psql_tsvector_string(model):
+        """
+        Provides the tsvector string for postgres defining which columns and
+        json fields to be used for full text search index.
+
+        Example:
+        s = create_tsvector_string('title', 'location')
+        s += " || ' ' || coalesce(((content ->> 'description')), '')"
+        s += " || ' ' || coalesce(((content ->> 'organizer')), '')"
+
+        :return: tsvector string
+        """
+
+        join = " || ' ' || "
+
+        # identify search columns
+        columns = [p for p in model.es_properties if p in model.__dict__
+                   and isinstance(model.__dict__[p], InstrumentedAttribute)]
+        s = Searchable.create_tsvector_string(*columns) if columns else ''
+
+        # identify content and meta properties
+        for p in model.es_properties:
+            if p in model.__dict__ and isinstance(model.__dict__[p],
+                                                  content.dict_property):
+                s += join if s else ''
+                s += f"coalesce((({model.__dict__[p].attribute} " \
+                     f"->> '{p}')), '')"
```

Comment:
Maybe we can simplify this by making `dict_property` a subclass of `hybrid_property` which implicitly generates the JSON access syntax.

Reply:
Also this could go wrong if the name of the attribute doesn't match the name of the column. So you should retrieve the column from the model using attribute, and use the column to generate the first part of the query, rather than trusting that the names match.

Reply:
I think I will try my hand at this. Rather than deriving from hybrid_property I can just extend dict_property to behave the same as a hybrid_property (i.e. returning a SQL expression when accessed from the class, rather than an instance), we won't be able to handle custom getters, without also defining a custom expression, but that's fine, in that case we can just fall back to what it did before.

Reply:
No problem, I am happy if you support here.

Reply:
the improved `dict_property` has been merged into master, so now you shouldn't need to do this second loop anymore, the first loop should be sufficient.

Reply:
second loop completely removed